### PR TITLE
Expose chains metrics in openshift operator for monitoring

### PR DESCRIPTION
**Pull request by sudhishmk**
# Changes

* Exposing chains metrics for operator in OpenShift environment, fixes: #1955 
* Related to the [PR](https://github.com/tektoncd/chains/pull/1034)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by khrm** [comment]
Please remove the draft from this.

**Comment by sudhishmk** [comment]
> Please remove the draft from this.

Done

**Review by savitaashture — COMMENTED** [review]
/lgtm

File: cmd/openshift/operator/kodata/openshift-monitoring/02-chains-monitoring.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: openshift-chains-read
+  namespace: tekton-chains
```
Comment (jkandasa) [inline cmd/openshift/operator/kodata/openshift-monitoring/02-chains-monitoring.yaml:20]: @sudhishmk Do we deploy chains on `tekton-chains` namespace.
I hope we do transform the namespace before applying.
however it is good to have the default namespace of OpenShift environment
that is `openshift-pipelines`.

@concaf @piyush-garg please confirm the namespace.

Comment (PuneetPunamiya) [reply cmd/openshift/operator/kodata/openshift-monitoring/02-chains-monitoring.yaml:20]: Yes we deploy chains in `openshift-pipelines` namespace in Openshift environment

Comment (sudhishmk) [reply cmd/openshift/operator/kodata/openshift-monitoring/02-chains-monitoring.yaml:20]: Thanks, will check that.

Comment (sudhishmk) [reply cmd/openshift/operator/kodata/openshift-monitoring/02-chains-monitoring.yaml]: Updated the namespace.

**Review by jkandasa — APPROVED** [review]

**Comment by jkandasa** [comment]
thanks @sudhishmk :tada:

**Comment by piyush-garg** [comment]
I am not sure if role and role binding is needed, is not only service monitor will do the job ?

**Comment by sudhishmk** [comment]
> I am not sure if role and role binding is needed, is not only service monitor will do the job ?

did you mean that since we already have role and rolebinding defined for pipeline/ trigger monitoring service, new role and role binding is not needed?

**Comment by piyush-garg** [comment]
> > I am not sure if role and role binding is needed, is not only service monitor will do the job ?
> 
> did you mean that since we already have role and rolebinding defined for pipeline/ trigger monitoring service, new role and role binding is not needed?

Yes, we can even remove for triggers too, and keep single

**Comment by sudhishmk** [comment]
> > > I am not sure if role and role binding is needed, is not only service monitor will do the job ?
> > 
> > 
> > did you mean that since we already have role and rolebinding defined for pipeline/ trigger monitoring service, new role and role binding is not needed?
> 
> Yes, we can even remove for triggers too, and keep single

Done.

**Pull request merged**